packages/notifi-frontend-client/lib/NotifiFrontendClient.ts

```diff
@@ -73,4 +88,49 @@ export class NotifiFrontendClient {
 
     await Promise.all([saveAuthorizationPromise, saveRolesPromise]);
   }
+
+  async beginLoginViaTransaction({
+    walletBlockchain,
+    walletAddress,
+  }: BeginLoginProps): Promise<BeginLogInByTransactionResult> {
+    const { dappAddress } = this._configuration;
+
+    const result = await this._service.beginLogInByTransaction({
+      walletAddress: walletAddress,
+      walletBlockchain: walletBlockchain,
+      dappAddress,
+    });
+
+    const nonce = result.beginLogInByTransaction.nonce;
+
+    this.clientRandomUuid = await window.crypto.randomUUID();
+
+    return { nonce };
+  }
+
+  async completeLoginViaTransaction({
+    walletBlockchain,
+    walletAddress,
+    transactionSignature,
+  }: CompleteLoginProps): Promise<CompleteLogInByTransactionMutation> {
+    const { dappAddress } = this._configuration;
+
+    if (this.clientRandomUuid == null) {
+      throw new Error(
+        'BeginLoginViaTransaction is required to be called first',
+      );
+    }
+
+    const result = await this._service.completeLogInByTransaction({
+      walletAddress: walletAddress,
+      walletBlockchain: walletBlockchain,
+      dappAddress,
+      randomUuid: this.clientRandomUuid,
```

Comment:
it's not an issue here, but in general, for async functions, it's good practice to store mutable variables in enclosing scope into a local `const` so the same value is used throughout the function's execution (between yields)

Reply:
We mentioned during our call that it is best to call `const clientRandomUuid = this._clientRandomUuid;` and use `clientRandomUuid` throughout the function so no one can update it before the other parts of the function fire off.